console/src/app/pages/projects/apps/app-create/app-create.component.html

```diff
@@ -395,31 +426,61 @@ <h1>{{ 'APP.PAGES.CREATE_DESC_TITLE' | translate }}</h1>
       <div class="content">
         <ng-container *ngIf="formappType?.value?.createType === AppCreateType.SAML">
           <div class="saml">
-            <cnsl-form-field class="formfield">
-              <cnsl-label>{{ 'APP.SAML.URL' | translate }}</cnsl-label>
-              <input cnslInput formControlName="metadataUrl" placeholder="https://" />
-            </cnsl-form-field>
-
-            <span class="cnsl-app-or cnsl-secondary-text">{{ 'APP.SAML.OR' | translate }}</span>
-
-            <input
-              #xmlFileInput
-              style="display: none"
-              class="file-input"
-              type="file"
-              (change)="onDropXML($any($event.target).files)"
-            />
-            <button type="button" mat-stroked-button (click)="$event.preventDefault(); xmlFileInput.click()">
-              {{ 'APP.SAML.XML' | translate }}
-            </button>
+            <div class="cnsl-secondary-text metadata-option">
+              <span>{{ 'APP.SAML.METADATAOPT1' | translate }}</span>
+              <form [formGroup]="samlConfigForm">
+                <cnsl-form-field class="formfield">
+                  <cnsl-label></cnsl-label>
+                  <input cnslInput formControlName="metadataUrl" placeholder="https://" />
+                </cnsl-form-field>
+              </form>
+            </div>
+
+            <div class="cnsl-secondary-text metadata-option">
+              <span>{{ 'APP.SAML.METADATAOPT2' | translate }}</span>
+
+              <form>
+                <input
+                  #xmlFileInput
+                  style="display: none"
+                  class="file-input"
+                  type="file"
+                  accept="text/xml,application/xml"
+                  (change)="onDropXML($any($event.target).files)"
+                />
+                <button
+                  type="button"
+                  class="upload-button"
+                  mat-stroked-button
+                  (click)="$event.preventDefault(); xmlFileInput.click()"
+                >
+                  {{ 'APP.SAML.UPLOAD' | translate }}
+                </button>
+              </form>
+            </div>
+
+            <div class="cnsl-secondary-text metadata-option">
+              <span>{{ 'APP.SAML.METADATAOPT3' | translate }}</span>
+              <form [formGroup]="samlConfigForm">
+                <cnsl-form-field class="formfield">
+                  <cnsl-label>{{ 'APP.SAML.ENTITYID' | translate }}</cnsl-label>
+                  <input cnslInput formControlName="entityId" placeholder="https://" (change)="changeEntitityIdOrAcsURL()" />
```

Comment:
```suggestion
                  <input cnslInput formControlName="entityId" placeholder="https://" />
```

Reply:
please use the `this.samlConfigForm.valueChanges.subscribe` in the ts file to react to changes so we can remove the `changeEntityIdOrAcsURL()`